spark_pipeline_framework/transformers/http_data_receiver/v4/http_data_receiver.py

```diff
@@ -0,0 +1,277 @@
+            self, "http_request_generator", ""
+        )
+        self._setDefault(http_request_generator=None)
+
+        self.response_processor: Param[RESPONSE_PROCESSOR_TYPE] = Param(
+            self, "response_processor", ""
+        )
+        self._setDefault(response_processor=None)
+
+        self.batch_count: Param[int] = Param(self, "batch_count", "")
+        self._setDefault(batch_count=None)
+
+        self.batch_size: Param[int] = Param(self, "batch_size", "")
+        self._setDefault(batch_size=None)
+
+        self.cache_storage_level: Param[Optional[StorageLevel]] = Param(
+            self, "cache_storage_level", ""
+        )
+        self._setDefault(cache_storage_level=None)
+
+        self.credentials: Param[OAuth2Credentails] = Param(self, "credentials", "")
+        self._setDefault(credentials=None)
+
+        self.auth_url: Param[str] = Param(self, "auth_url", "")
+        self._setDefault(auth_url=None)
+
+        self.run_sync: Param[bool] = Param(self, "run_sync", "")
+        self._setDefault(run_sync=None)
+
+        self.raise_error: Param[bool] = Param(self, "raise_error", "")
+        self._setDefault(raise_error=None)
+
+        kwargs = self._input_kwargs
+        self.setParams(**kwargs)
+
+    def _transform(self, df: DataFrame) -> DataFrame:
+        # Setting the variables
+        name: str = self.getOrDefault(self.name)
+        view_name: str = self.getOrDefault(self.view_name)
+        error_view: str = self.getOrDefault(self.error_view)
+        batch_count: int = self.getOrDefault(self.batch_count)
+        batch_size: int = self.getOrDefault(self.batch_size)
+        http_request_generator: REQUEST_GENERATOR_TYPE = self.getOrDefault(
+            self.http_request_generator
+        )
+        response_processor: RESPONSE_PROCESSOR_TYPE = self.getOrDefault(
+            self.response_processor
+        )
+        cache_storage_level: Optional[StorageLevel] = self.getOrDefault(
+            self.cache_storage_level
+        )
+        credentials: Optional[OAuth2Credentails] = self.getOrDefault(self.credentials)
+        auth_url: Optional[str] = self.getOrDefault(self.auth_url)
+        run_sync: bool = self.getOrDefault(self.run_sync)
+        raise_error: bool = self.getOrDefault(self.raise_error)
+        progress_logger: Optional[ProgressLogger] = self.getProgressLogger()
+
+        with ProgressLogMetric(
+            name=f"{name}_http_data_receiver(V4)", progress_logger=progress_logger
+        ):
+            requests_df: DataFrame = create_empty_dataframe(
+                df.sparkSession,
+                StructType(
+                    [
+                        StructField("url", StringType()),
+                        StructField("headers", StringType()),
+                        StructField("state", StringType()),
+                    ]
+                ),
+            )
+            for requests in chunked(http_request_generator(df), 1000):
```

Comment:
test if chunked will work as intended on this generator. What will happen if there are < 1000 requests available and generator raises StopIteration?

Reply:
why is this hardcoded as 1000?